### DISCUSSION TITLE
2B/aspire style maps have too high star rating

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#18045] [Ideas]
Some maps that are 2B or aspire-like seem to have a star rating that doesn't reflect their difficulty at all. While issues with those maps, being extreme cases and all, are not uncommon, I remembered them getting displayed better in the past.
As it turns out, [this PR](https://github.com/ppy/osu/pull/14845) seems to be responsible for *most* of the star rating issues I've seen in these specific maps. I did a bisect, specifically looking for the change from ~3 stars to ~9 stars in [this map](https://osu.ppy.sh/beatmapsets/90935#osu/246410), which resulted in
```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
0292fe1c834ebf68f22f85832a776b6740030865
0f248d55107e574876d139b2d030f98575f62704
4edf559408e2cfe48bcc805d61cf914d032bf839
200149c9d7203084a08d77303b8091a3634aea8d
30eb08d394760c13f838363d4f58fcf7833991f9
14d405786ee50ac4a641717dec83a5a11e092b2c
We cannot bisect more!
```
Notable, cec063206e5deadab3fd39892bf061d05343b795 is fine in that case, while 14d405786ee50ac4a641717dec83a5a11e092b2c is "wrong" (all other commits generated a star rating value of 0 stars for me).

Some data for a selection of maps (star rating change from cec063206e5deadab3fd39892bf061d05343b795 to 14d405786ee50ac4a641717dec83a5a11e092b2c):
[Mashed Potatoes](https://osu.ppy.sh/beatmapsets/474413#osu/1013379), difficulty `2B`: 4.71 -> 10.84
[Mayday (feat. Laura Brehm)](https://osu.ppy.sh/beatmapsets/756794#osu/1605148): 5.55 -> 13.12
[cherry blossoms explode across the dying horizon](https://osu.ppy.sh/beatmapsets/1236902#osu/2573886), difficulty `
wabi • sabi`: 9.30 -> 17.83, difficulty `‡debug`: 9.82 -> 37.22
[Unshakable](https://osu.ppy.sh/beatmapsets/1237363#osu/2571858): 12.98 -> 38.91
[The Solace of Oblivion](https://osu.ppy.sh/beatmapsets/594751#osu/1257904): 110.59 -> 53.39
[Der Wald (Kors K Remix)](https://osu.ppy.sh/beatmapsets/537048#osu/1529757), difficulty `Hey jia`: 15.19 -> 24.36, difficulty `NiNo's Aspire`: 54.78 -> 10.73
[Aleph-0](https://osu.ppy.sh/beatmapsets/791900#osu/1680610), difficulty `Omega`: 9.55 -> 17.12, difficulty `Omega + 1`: 9.60 -> 19.14
[Endless Tewi-ma Park](https://osu.ppy.sh/beatmapsets/90935#osu/272317), difficulty `Tewi 2B`: 2.79 -> 9.20, difficulty `Tewi 2B Expert Edition`: 4.74 -> 11.56

All of these retain this high star rating (or even higher than at the specified commit) as of version 2022.501.0.

I don't think this is a huge issue for now, but I wanted to mention it at the very least, as I believe this should be fixed or improved upon eventually (stable generates relatively sane star rating values for all these maps).

**Comment by bdach** [comment]
From what I recall it has been expressed several times that we are fine with aspire maps having a broken star rating if it results in a better overall pp system. And I continue to hold this view, personally. Maps that contain game breaking mechanics can also break star rating, and that's fine.
